common/cpp/src/google_smart_card_common/value_debug_dumping_unittest.cc

```diff
@@ -0,0 +1,193 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <google_smart_card_common/value_debug_dumping.h>
+
+#include <stdint.h>
+
+#include <limits>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <google_smart_card_common/formatting.h>
+#include <google_smart_card_common/logging/hex_dumping.h>
+#include <google_smart_card_common/unique_ptr_utils.h>
+#include <google_smart_card_common/value.h>
+
+namespace google_smart_card {
+
+TEST(ValueDebugDumpingTest, Null) {
+  EXPECT_EQ(DebugDumpValueSanitized(Value()), Value::kNullTypeTitle);
+  EXPECT_EQ(DebugDumpValueFull(Value()), Value::kNullTypeTitle);
+}
+
+TEST(ValueDebugDumpingTest, Boolean) {
+  const std::string full_false = DebugDumpValueFull(Value(false));
+  const std::string full_true = DebugDumpValueFull(Value(true));
+  EXPECT_EQ(full_false, "false");
+  EXPECT_EQ(full_true, "true");
+
+  const std::string sanitized_false = DebugDumpValueSanitized(Value(false));
+  const std::string sanitized_true = DebugDumpValueSanitized(Value(true));
+#ifdef NDEBUG
+  EXPECT_EQ(sanitized_false, Value::kBooleanTypeTitle);
+  EXPECT_EQ(sanitized_true, Value::kBooleanTypeTitle);
+#else
+  EXPECT_EQ(sanitized_false, full_false);
+  EXPECT_EQ(sanitized_true, full_true);
+#endif
+}
+
+TEST(ValueDebugDumpingTest, Integer) {
+  constexpr int64_t kInteger = 123;
```

Comment:
nit: Again I think that the expected results should be hardcoded here, making the expectations explicit and allowing to catch errors in the underlying functions.

Reply:
(Here and in other places in this file)